Integrations/JsonWhoIs/JsonWhoIs.py

```diff
@@ -0,0 +1,186 @@
+
+''' HELPER FUNCTIONS '''
+@logger
+def http_request(method, url_suffix, params=None, max_retry=3):
+    # A wrapper for requests lib to send our requests and handle requests and responses better
+    res = requests.request(
+        method,
+        BASE_URL + url_suffix,
+        verify=USE_SSL,
+        params=params,
+        data=None,
+        headers=HEADERS
+    )
+    # Handle error responses gracefully
+    if res.status_code not in {200}:
+        if max_retry > 0:
+            max_retry -= 1
+            return http_request(method=method,
+                                url_suffix=url_suffix,
+                                params=params,
+                                max_retry=max_retry)
+        raise Exception(f'Error enrich url with JsonWhoIS API, status code {res.status_code}')
+    return res.json()
+
+
+def dict_by_ec(cur_dict: dict):
+    """ Create dict (Json) by entry contexts convention
+    Capitalize first char, remove nulls
+    :param cur_dict: dictionary
+    :return: dictionary by conventions
+    """
+    if not cur_dict:
+        return None
+    return {key.capitalize(): cur_dict[key] for key in cur_dict if cur_dict[key]}
+
+
+def list_by_ec(cur_list: list, key_needed):
+    """ Create list of dict (Json) by entry contexts convention
+    Capitalize first char in dict, remove nulls, remove not needed parameters
+    :param cur_list: list of dict
+    :param key_needed: key to save
+    :return: modified list by description above
+    """
+    if not cur_list:
+        return None
+    cur_list = [createContext(index, removeNull=True) for index in cur_list]
+
+    def cur_ec(index):
+        return {key.capitalize(): index[key] for key in index if key in key_needed}
+    cur_list = [cur_ec(contact) for contact in cur_list]
+    return cur_list
+
+
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+
+@logger
+def whois(url: str) -> tuple:
+    """Get Rest API raw from JsonWhoIs service
+    :param url: url to search on
+    :return: dict object of JsonWhoIs service
+    """
+    # Perform request
+    params = {
+        'domain': url
+    }
+    raw = http_request(method='GET',
+                       url_suffix='/api/v1/whois',
+                       params=params)
+
+    # entry context by code convention
+    ec_temp = {
+        'DomainStatus': raw.get('status'),
+        'NameServers': list_by_ec(raw.get('nameservers'), key_needed=['name']),
+        'CreationDate': raw.get('created_on'),
+        'UpdatedDate': raw.get('updated_on'),
+        'ExpirationDate': raw.get('expires_on'),
+        'Registrar': dict_by_ec(raw.get('registrar')),
+        'Registrant': list_by_ec(raw.get('registrant_contacts'), key_needed=['name', 'phone', 'email']),
+        'Admin': list_by_ec(raw.get('admin_contacts'), key_needed=['name', 'phone', 'email'])
+    }
+
+    # Clear unused fields
+    ec_temp = createContext(ec_temp, removeNull=True)
+
+    # Build all ec
+    ec = {
+        'Domain': {
+            'WHOIS': ec_temp
```

Comment:
add those two levels in the `return_outputs` call in `whois_command` to avoid creating shortcuts